linkedList.js

```diff
@@ -3,6 +3,112 @@
  * @description  Main Module
  * @return {Object} an object exposing methods to be used to manipulate a linked list
  */
-function linkedListGenerator(){
+function linkedListGenerator() {
 
-}
\ No newline at end of file
+  // Declare let for head and tail
+  let head = null;
+  let tail = null;
+
+  function getHead() {
+    return head;
+  };
+
+  function getTail() {
+    return tail;
+  };
+
+  //Helper function for a newNode
+  function newNode1(val) {
+    return {
+      value: val,
+      next: null,
+    }
+  };
+
+  function add(val) {
+    let newNode = newNode1(val); // assign variable for newNode function
+
+    if (head === null) { // head is a thing, so keep going
+      head = newNode; // assign global variable head to newNode
+      tail = newNode; // assign global variable tail to newNode in order to manipulate next tag
+    } else {
+      let current = head; // assign current to head (which was newNode)
+      while (current.next !== null) {
+        current = current.next;
+      }
+      current.next = newNode; // update current tag is now attached to newNode
+      tail = newNode; // update linked list to point tail to newNode
+    }
+    return newNode; // 
+  };
+
+  function get(num) {
+    let current = head;
+    let index = 0;
+
+    while (current !== null) {
+      if (index === num) {
+        return current;
+      } else {
+        current = current.next;
+        index++;
+      }
+    }
+    return false;
+  };
+
+  function remove(num) {
+    let current = head; // assign variable to a current, which always starts at head
+    let index = 0;
+    let prev = null;
+
+    while (current !== null) {
+      if (index === num) { // found index, remove current node by assigning to prev node
```

Comment:
So it looks like this if statement is probably keeping your while loop from actually progressing unless the 'num' value is 0. I don't think this if statement is needed at all.